Amazon.QLDB.Driver/session/QldbSession.cs

```diff
@@ -124,78 +118,45 @@ public QldbSession Renew()
         /// <exception cref="AmazonServiceException">Thrown when there is an error executing against QLDB.</exception>
-        public T Execute<T>(Func<TransactionExecutor, T> func, Action<int> retryAction)
+        public T Execute<T>(Func<TransactionExecutor, T> func)
         {
             ValidationUtils.AssertNotNull(func, "func");
             this.ThrowIfDisposedOrClosed();
 
-            var executionAttempt = 0;
-            while (true)
+            ITransaction transaction = null;
+            try
             {
-                ITransaction transaction = null;
-                try
-                {
-                    transaction = this.StartTransaction();
-                    T returnedValue = func(new TransactionExecutor(transaction));
-                    if (returnedValue is IResult)
-                    {
-                        returnedValue = (T)(object)BufferedResult.BufferResult((IResult)returnedValue);
-                    }
-
-                    transaction.Commit();
-                    return returnedValue;
-                }
-                catch (InvalidSessionException ise)
+                transaction = this.StartTransaction();
+                T returnedValue = func(new TransactionExecutor(transaction));
+                if (returnedValue is IResult)
                 {
-                    this.isClosed = true;
-                    throw ise;
+                    returnedValue = (T)(object)BufferedResult.BufferResult((IResult)returnedValue);
                 }
-                catch (TransactionAbortedException ae)
-                {
-                    throw ae;
-                }
-                catch (OccConflictException oce)
-                {
-                    if (executionAttempt >= this.retryLimit)
-                    {
-                        throw oce;
-                    }
 
-                    this.logger.LogInformation("Retrying the transaction. {msg}", oce.Message);
-                }
-                catch (TransactionAlreadyOpenException taoe)
-                {
-                    this.NoThrowAbort(transaction);
-                    if (executionAttempt >= this.retryLimit)
-                    {
-                        throw taoe.InnerException;
-                    }
-
-                    this.logger.LogInformation("Retrying the transaction. {msg}", taoe.Message);
-                }
-                catch (AmazonQLDBSessionException aqse)
-                {
-                    this.NoThrowAbort(transaction);
-                    if (executionAttempt >= this.retryLimit ||
-                        ((aqse.StatusCode != HttpStatusCode.InternalServerError) &&
-                        (aqse.StatusCode != HttpStatusCode.ServiceUnavailable)))
-                    {
-                        throw aqse;
-                    }
+                transaction.Commit();
+                return returnedValue;
+            }
```

Comment:
I see in Node, it is catching `TransactionAbortedException`. As the name implies, we should catch this exception and call `this.NoThrowAbort()`

Reply:
this.NoThrowAbort() is called inside StartTransaction() where TransactionAbortedException was thrown. I changed it this way to simplify the logic.

Reply:
TransactionAbortedException is only thrown when user calls txn.abort() within a lambda, see TransactionExecutor.cs

Reply:
Yes, you are right. I will fixed it in the codecommit pr